pkg/controller/syncsetinstance/syncsetinstance_controller.go

```diff
@@ -303,6 +312,19 @@ func (r *ReconcileSyncSetInstance) Reconcile(request reconcile.Request) (reconci
 	return reconcile.Result{RequeueAfter: reapplyDuration}, nil
 }
 
+func (r *ReconcileSyncSetInstance) hasSourceSecretsInAnotherNamespace(ssi *hivev1.SyncSetInstance, spec *hivev1.SyncSetCommonSpec) bool {
+	// This check is not relevant for SelectorSyncSets
+	if ssi.Spec.SyncSetRef == nil {
+		return false
+	}
+	for _, secret := range spec.Secrets {
+		if secret.SourceRef.Namespace != ssi.Namespace {
```

Comment:
Allow empty namespace.

Reply:
Done, didn't add a unit test in this spot because the harness doesn't seem to like an empty namespace and I don't really want to get into that right now.